### PR TITLE
Add-teams-dynamic-scripts

### PR DESCRIPTION
**Pull request by rschouten97**

### CODE REVIEW

File: dynamicPermission.Teams.ps1 [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+$InformationPreference = "Continue"
+$WarningPreference = "Continue"
+
+# Used to connect to Azure AD Graph API
+$AADtenantID = $c.AADtenantID
+$AADAppId = $c.AADAppId
+$AADAppSecret = $c.AADAppSecret
+
+# Hardcoded Object ID of Azure AD User to set as owner of team
+$ownerAADUserId = "12345678-abcd-efgh-ijkl-mnopqrstuvwxyz90"
```
Comment (mouki9) [inline dynamicPermission.Teams.ps1]: line 29, $ownerAADUserId is hardcoded

File: resources.Teams.ps1 [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+$InformationPreference = "Continue"
+$WarningPreference = "Continue"
+
+# Used to connect to Azure AD Graph API
+$AADtenantID = $c.AADtenantID
+$AADAppId = $c.AADAppId
+$AADAppSecret = $c.AADAppSecret
+
+# Hardcoded Object ID of Azure AD User to set as owner of team
+$ownerAADUserId = "12345678-abcd-efgh-ijkl-mnopqrstuvwxyz90"
```
Comment (mouki9) [inline resources.Teams.ps1:29]: hardcoded $ownerAADUserId

File: resources.Teams.ps1 [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+                    )
+                }
+
+                if (-Not($dryRun -eq $True)) {
+                    $headers = New-AuthorizationHeaders -TenantId $AADtenantID -ClientId $AADAppId -ClientSecret $AADAppSecret
+
+                    $body = $teamObject | ConvertTo-Json -Depth 10
+                    $baseUri = "https://graph.microsoft.com/"
+                    $splatWebRequest = @{
+                        Uri     = "$baseUri/v1.0/teams"
```
Comment (mouki9) [inline resources.Teams.ps1:242]: for this specific endpoint permissions need to be set correctly. Please add this to the readme.

**Review by mouki9 — CHANGES_REQUESTED** [review]
See comments in resources.Teams.ps1

File: resources.Teams.ps1 [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+$InformationPreference = "Continue"
+$WarningPreference = "Continue"
+
+# Used to connect to Azure AD Graph API
+$AADtenantID = $c.AADtenantID
+$AADAppId = $c.AADAppId
+$AADAppSecret = $c.AADAppSecret
+
+# Hardcoded Object ID of Azure AD User to set as owner of team
+$ownerAADUserId = "12345678-abcd-efgh-ijkl-mnopqrstuvwxyz90"
```
[the remark this answers is not in the document]
Comment (rschouten97) [reply resources.Teams.ps1:29]: Added to config

File: resources.Teams.ps1 (hunk 3 above)
Comment (rschouten97) [reply resources.Teams.ps1:242]: Added to readme

File: resources.Teams.ps1 [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+            Write-Verbose "Querying Azure AD group that matches filter '$($filter)'"
+
+            $baseUri = "https://graph.microsoft.com/"
+            $splatWebRequest = @{
+                Uri     = "$baseUri/v1.0/groups?`$filter=$($filter)&`$count=true"
+                Headers = $headers
+                Method  = 'GET'
+            }
+            $currentTeam = $null
+            $currentTeamResponse = Invoke-RestMethod @splatWebRequest -Verbose:$false
```
[the remark this answers is not in the document]
Comment (rhouthuijzen) [reply resources.Teams.ps1:200]: We agree. Will be reworked in newer version of the connector.

**Review by rhouthuijzen — APPROVED** [review]
Approved

**Review by rhouthuijzen — APPROVED** [review]
Approved

**Review by mouki9 — APPROVED** [review]

**Pull request merged**